### ISSUE TITLE
set d.scores['cosine similarity'] = 0.96 and use flow.post to request it

### ISSUE THREAD

**Issue by 752377941** [jina-ai/jina#2861]
**Describe the bug**
when I set d.scores['cosine similarity'] = 0.96 and use flow.post to request it,then it raise

```ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/uvicorn/protocols/http/httptools_impl.py", line 371, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/uvicorn/middleware/proxy_headers.py", line 59, in __call__
    return await self.app(scope, receive, send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/fastapi/routing.py", line 233, in app
    is_coroutine=is_coroutine,
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/fastapi/routing.py", line 129, in serialize_response
    raise ValidationError(errors, field.type_)
pydantic.error_wrappers.ValidationError: 4 validation errors for RequestProto
response -> data -> docs -> 0 -> matches -> 0 -> scores -> values
  value is not a valid list (type=type_error.list)
response -> data -> docs -> 0 -> matches -> 1 -> scores -> values
  value is not a valid list (type=type_error.list)
response -> data -> docs -> 0 -> matches -> 2 -> scores -> values
  value is not a valid list (type=type_error.list)
response -> data -> docs -> 0 -> matches -> 3 -> scores -> values
  value is not a valid list (type=type_error.list)
Traceback (most recent call last):
  File "app.py", line 134, in <module>
    main()
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "app.py", line 127, in main
    index(num_docs, reindex)
  File "app.py", line 78, in index
    on_done=ppr,
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/jina/clients/mixin.py", line 66, in post
    **kwargs,
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/jina/helper.py", line 1102, in run_async
    return asyncio.run(func(*args, **kwargs))
  File "/usr/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/jina/clients/mixin.py", line 49, in _get_results
    async for resp in c._get_results(*args, **kwargs):
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/jina/clients/base/http.py", line 84, in _get_results
    r_status, r_str = await resp
  File "/usr/lib/python3.7/asyncio/tasks.py", line 533, in _wait_for_one
    return f.result()  # May raise f.exception().
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/jina/clients/base/http.py", line 22, in _get_http_response
    resp_str = await response.json()
  File "/home/zhiyu/.local/share/virtualenvs/jina_examples-JUjaqce_/lib/python3.7/site-packages/aiohttp/client_reqrep.py", line 1103, in json
    headers=self.headers,
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/plain; charset=utf-8', url=URL('http://0.0.0.0:45678/post')
```

---

**Environment**
jina v2.0.2

**Comment by JoanFM** [comment]
can u share some more detail on ur code, flow and executors? it does not seem that the error is related with the scores?

Where is the scores set, at the client? at the executor?

**Comment by 752377941** [comment]
> can u share some more detail on ur code, flow and executors? it does not seem that the error is related with the scores?
> 
> Where is the scores set, at the client? at the executor?

at the executor, it has been solved at v2.0.3.

**Comment by 752377941** [comment]
> can u share some more detail on ur code, flow and executors? it does not seem that the error is related with the scores?
> 
> Where is the scores set, at the client? at the executor?

When I set scores, the RequestProto
